### ISSUE TITLE
Crest v0.10.0 is buggy (?)

### ISSUE THREAD

**Issue by mamantoha** [mamantoha/crest#86]
[shards-info](https://github.com/mamantoha/shards-info) with `crest` `v0.10.1` randomly crash on Heroku about 1-2 times per hour :shit:

**Comment by mamantoha** [comment]
https://github.com/mamantoha/crest/compare/73c1d71936fc131c9251c66ba018935f91c8284f...mamantoha:9a07f928a5829850c75f5149c6b96d03b505600c#diff-3994e7f5f940813c595702c024b0e5f7R48

**Comment by mamantoha** [comment]
Crystal code to reproduce the issue:

```crystal
require "http/client"
require "uri"

url = "https://api.github.com"
uri = URI.parse(url)
path = "/repos/crystal-lang/crystal"

headers = HTTP::Headers.new
headers.add("Accept", "application/vnd.github.mercy-preview+json")

client = HTTP::Client.new(uri)

resp = client.exec("GET", path, headers: headers)
puts resp.body

puts "Wait at least 1 minute and then press Enter"
gets

resp = client.exec("GET", path, headers: headers)
puts resp.body
```

**Comment by mamantoha** [comment]
Adding this line fixes this issue:

```crystal
client.compress = false
```
:tada:

**Comment by mamantoha** [comment]
Still exists in Crystal 0.25 :crying_cat_face:

**Comment by mamantoha** [comment]
Still exists in Crystal 0.26 :crying_cat_face:

**Comment by mamantoha** [comment]
Still exists in Crystal 0.29 :crying_cat_face:

**Comment by wontruefree** [comment]
should this issue be upstreamed to the crystal project?

**Comment by mamantoha** [comment]
Still exists in Crystal 1.1.0 :crying_cat_face:

**Comment by cyangle** [comment]
It seems like this has been fixed with the docker nightly build.

**Comment by mamantoha** [comment]
@cyangle still exists

```
crystal -v                
Crystal 1.3.0-dev [6ce66047c] (2021-12-07)

LLVM: 10.0.0
Default target: x86_64-unknown-linux-gnu
```

**Comment by cyangle** [comment]
@mamantoha Could you try it out with latest crystallang/crystal:nightly-alpine? It works for me.

**Comment by cyangle** [comment]
This bug seems to be fixed by https://github.com/crystal-lang/crystal/pull/12536

**Comment by mamantoha** [comment]
Fixed in Crystal 1.6.0 😸 🎉
